src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -371,16 +374,24 @@ public Builder returns(Type returnType) {
     public Builder addParameters(Iterable<ParameterSpec> parameterSpecs) {
       checkArgument(parameterSpecs != null, "parameterSpecs == null");
       for (ParameterSpec parameterSpec : parameterSpecs) {
+        addParameterJavadoc(parameterSpec);
         this.parameters.add(parameterSpec);
       }
       return this;
     }
 
     public Builder addParameter(ParameterSpec parameterSpec) {
+      addParameterJavadoc(parameterSpec);
       this.parameters.add(parameterSpec);
       return this;
     }
 
+    private void addParameterJavadoc(ParameterSpec parameterSpec) {
+      if (!parameterSpec.javadoc.isEmpty()) {
+        this.parameterJavadoc.add("@param $L $L", parameterSpec.name, parameterSpec.javadoc);
```

Comment:
Is there any reason to do this eagerly? Why not aggregate when we emit and avoid all this code?